src/Lean/Elab/MutualDef.lean

```diff
@@ -169,7 +169,7 @@ private def expandWhereStructInst : Macro
       | `(letDecl|$decl:letIdDecl)   => pure decl
       | _                               => Macro.throwUnsupported
     let structInstFields ← letIdDecls.mapM fun
-      | stx@`(letIdDecl|$id:ident $binders* $[: $ty?]? := $val) => withRef stx do
+      | stx@`(letIdDecl|$id:ident $binders:letIdBinder* $[: $ty?]? := $val) => withRef stx do
```

Comment:
This is the more problematic issue: a single `letIdBinder` also accepts a list of idents (or holes), so by the new semantics `$binders*` is interpreted as such because the "content parse" `$binders:ident*` consumes one more token than the "antiquotation parse" `$binders:letIdBinder` without the star. This is because the star is technically not part of the antiquotation but a separate modifier parsed afterwards. Note also that while this use site failed to compile, others like
```
  | `(have $x $bs* $[: $type]? := $val; $body)            => `(let_fun $x $bs* $[: $type]? := $val; $body)
```
are still type-correct, but now are more specific than should be, breaking many tests.

I think this points to a weakness in our grammar: we should avoid nesting sequences using the same separator (none, in this case):
```
letIdLhs ::= ident letIdBinder* ...
letIdBinder ::= simpleBinderWithoutType | ...
simpleBinderWithoutType ::= binderIdent*
```
I think we should replace `simpleBinderWithoutType` with a single `binderIdent` to avoid this inherent garmmar ambiguity. This would break compatibility with `simpleBinder` as noted in a comment above the former's definition, but hopefully these parts are already covered by `TSyntax` and thus simple to refactor.

Reply:
`simpleBinder`, and perhaps others, should probably be changed as well, but I think this would be a change for the better: currently we accept the following example inspired by @gebner's cursed juxtaposition:
```
#check fun a : Type 0f => _  -- fun (a : Type 0) (f : _) => _
```